### DISCUSSION TITLE
Connection help Nilan VP18cCom

### DISCUSSION THREAD

**Discussion by jyuenunaya** [veista/nilan#57] [Q&A] [answered]
Hi,

I am new on Home assistant and i have a nilan device in my home, a VP18cComp with a MODBUS LAN interface from Nilan Type 238481 (ES1140).

I try to connect after installing HACS and your github repository but with no result for now. So i have 2 questions :

What is the device name ? The modbus id ?
How can i find the modbus version of the system ?
The software version on my Nilan is : 1.6.56.0 and the controller is a CTS602 HMI

My version oh HA:
Home Assistant 2023.3.4
Supervisor 2023.03.1
Operating System 9.5

Return error:
Logger: pymodbus.logging
Source: custom_components/nilan/config_flow.py:59
Integration: Nilan (documentation, issues)
First occurred: 11:25:41 (1 occurrences)
Last logged: 11:25:41

Connection to (hidden IP, 502) failed: timed out

Many thanks for your help and your work !!

**Comment by veista** [comment]
Device name is what you like. Modbus id is by default 30, but you can see it in the HMI menu if it is something else. You can only find the modbus version by reading it via modbus. You can follow this  #9 to find out the modbusversion from chart. Yours is 22 so the newest I know.

**Comment by jyuenunaya** [comment, marked as answer]
Hi, 

Thank you for your answer !

I took a look in the HMI service menu and the modbus is 30 indeed.

Thanks, i understand so the modbus version is 22 because of the software version is 1.6.56.0.

So, have you any idea about this return error ? I always have the same error : Connection to (hidden IP, 502) failed: timed out

Ping to the IP is ok.

If i understood well, your code support the modbus version 22, i understand well ?

Thank you in advance for your help !

**Reply by veista** [reply]
The integration supports all fw versions of cts602.

What error does the integration give when you try to configure? Can you give a screenshot? Can you also take a picture of your devices type plate?

Also additional information, for example a picture of your setup.

**Reply by veista** [reply]
The error comes from,  not getting a modbus response from your device. If you are communicating with the device by other means it may cause this issue. Also if you have an incompatible modbus bridge.

**Reply by veista** [reply]
You might also have a different port on your bridge device.

**Reply by veista** [reply]
Ah. Now I understand. You are using the Nilan mobile gateway. That is NOT compatible with this integration. You will have to replace it with a supported rtu to tcp bridge.

**Reply by jyuenunaya** [reply]
> Ah. Now I understand. You are using the Nilan mobile gateway. That is NOT compatible with this integration. You will have to replace it with a supported rtu to tcp bridge.

Hi, 

Thanks for your explanation ! For now, i don't want to change this gateway because it's a recent installation.

When i will have more time, i will search for connect this gateway, maybe we have to add a auth part.

Thanks again and have a good day !
